toto/models/layout.py

```diff
@@ -9,23 +43,63 @@
 
 @attr.s(repr=False)
 class Layout(models__common.Signable):
+  """
+  The layout specifies each of the different steps and the requirements for
+  each step, as well as the public keys functionaries used to perform these
+  steps.
+
+  The layout also specifies additional steps called inspections
+  that are carried out during the verification.
+
+  Both steps and inspections can list rules that define how steps are
+  interconnected by their materials and products.
+
+  Layouts define a software supply chain and can be signed, dumped to a file,
+  and instantiated from a file.
+
+  <Attributes>
+    steps:
+        a list of Step objects
+
+    inspect:
+        a list of Inspection objects
+
+    keys:
+        a list of public keys used to verify the signature of link
+        metadata file related to a step
+
+    expires:
+        the expiration date of a layout
+
+  Todo:
+    - Warn if someone tries to add the private portion of a key to the keys
+
+  """
 
   _type = attr.ib("layout", init=False)
   steps = attr.ib([])
   inspect = attr.ib([])
   keys = attr.ib([])
   expires = attr.ib("")
```

Comment:
Maybe it's worth moving this to a datetime object with a default timeout (e.g., a month).